src/core/model_config_utils.cc

```diff
@@ -905,4 +919,75 @@ CheckAllowedModelOutput(
   return Status::Success;
 }
 
+#ifdef TRTIS_ENABLE_GPU
+Status
+CheckGPUCompatibility(const int gpu_id)
+{
+  // Query the compute capability from the device
+  cudaDeviceProp cuprops;
+  cudaError_t cuerr = cudaGetDeviceProperties(&cuprops, gpu_id);
+  if (cuerr != cudaSuccess) {
+    return Status(
+        RequestStatusCode::INTERNAL,
+        "unable to get CUDA device properties for GPU ID" +
+            std::to_string(gpu_id) + ": " + cudaGetErrorString(cuerr));
+  }
+
+  // A pair of integers to store the major and the minor of minimum supported
+  // gpu compute capability.
+  std::pair<int, int> min_compute_capability;
+  // Validates and populates the minimum supported Compute Capability by the
+  // Server.
+  Status status = ParseMinComputeCapability(min_compute_capability);
+  if (!status.IsOk()) {
+    return status;
+  }
+
+  if ((cuprops.major > min_compute_capability.first) ||
+      ((cuprops.major == min_compute_capability.first) &&
+       (cuprops.minor >= min_compute_capability.second))) {
+    return Status::Success;
+  } else {
+    return Status(
+        RequestStatusCode::UNSUPPORTED,
+        "The compute capability of gpu id " + std::to_string(gpu_id) + " is '" +
+            std::to_string(cuprops.major) + "." +
+            std::to_string(cuprops.minor) +
+            "' which is lesser than the minimum supported of '" +
+            TRTIS_MIN_COMPUTE_CAPABILITY + "' by TRTIS.");
+  }
+}
+
+Status
+ParseMinComputeCapability(std::pair<int, int>& min_compute_capability)
```

Comment:
Use std::stof instead of custom parser. Remember to check for failure and report error if invalid compute compatibility string.